core/blockchain/nullchain/nullchain.go

```diff
@@ -187,22 +188,35 @@ func (n *NullBlockchain) processBlock() {
 		n.log.Debugf("processing block %d with %d transactions", n.blockHeight, len(n.pending))
 	}
 
+	// prepare it first
+	ctx := context.Background()
```

Comment:
I'd prefer we use the main context here, but not sure if it matters all that much

Reply:
Core doesn't use the context in the `PrepareProposal` atm. In-real-life it would come from tendermint and I don't know what stuff it throw in it so that we can emulate it, and also the nullchain doesn't have a main context that it uses throughout so I think I'll leave it as is for now.